### PR TITLE
Clarify `client_id` authentication variables

### PR DESCRIPTION
**Pull request by fleischie**
This PR includes several related clarifications:
- Reword `client_id` authentication variable,
- reword `imap_id_retain` settings variable, and
- add relevant links between `imap_id_retain`, `imap_id_log` (from settings/core) and `client_id` auth varibale (from config_file/config_variable).

DOV-4940

### CODE REVIEW

File: source/settings/core.rst [hunk 1]
```diff
@@ -2301,6 +2301,10 @@ Variable Long name      Description
 -        passdb:<name>  Return passdb extra field "name". %{passdb:name:default} returns "default" if "name" doesn't exist (not returned if name exists but is empty) (v2.2.19+)
 ======== =============  =====================================================================================================
 
+Note: The ``client_id`` is only useful when proxying and needs to have the
+:ref:`setting-imap_id_retain` variable enabled. For logging this field on the
+client directly use :ref:`setting-imap_id_log`.
```
Comment (sirainen) [inline source/settings/core.rst:2306]: The client_id can be useful for non-proxying purposes also. Also I started wondering about this setting in general and did somewhat bigger cleanups: https://github.com/dovecot/documentation/pull/363

So this text could just be removed here.

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/config_file/config_variables.rst [hunk 2]
```diff
@@ -352,8 +352,11 @@ Authentication variables:
 |          |                       |                                                               |
 |          |                       | .. versionadded:: v2.2.19                                     |
 +----------+-----------------------+---------------------------------------------------------------+
-|          | client_id             | Expands to client ID request as IMAP arglist. Needs           |
-|          |                       | imap_id_retain=yes                                            |
+|          | client_id             | If :ref:`setting-imap_id_retain` is enabled this variable is  |
+|          |                       | populated with the client ID request as IMAP arglist. This is |
+|          |                       | mainly useful for logging the IMAP ID command when proxying.  |
```
[the remark this answers is not in the document]
Comment (fleischie) [reply source/configuration_manual/config_file/config_variables.rst]: Removed sentence.

File: source/settings/core.rst [hunk 3]
```diff
@@ -1456,15 +1456,17 @@ Example Setting:
...
+forward the IMAP ID command provided by the client.
 
 Example Setting:
 
 .. code-block:: none
 
      imap_id_retain=yes
 
+See :ref:`Auth variables <variables-auth>` for details on the ``client_id``
+field.
```
[the remark this answers is not in the document]
Comment (fleischie) [reply source/settings/core.rst]: Moved and reworded.

File: source/settings/core.rst (hunk 1 above)
Comment (fleischie) [reply source/settings/core.rst]: Removed.

**Review by vmo69 — APPROVED** [review]

**Review by sirainen — APPROVED** [review]

**Pull request merged**